src/App.tsx

```diff
@@ -1,14 +1,54 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
-import { TodoFilter } from './components/TodoFilter';
+import { TodoFilter, SelectOptions } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+const FILTER_MAP = {
+  [SelectOptions.All]: () => true,
+  [SelectOptions.Active]: (todo: Todo) => !todo.completed,
+  [SelectOptions.Completed]: (todo: Todo) => todo.completed,
+};
+
+function applyFilter(arr: Todo[], selectedOption: SelectOptions): Todo[] {
+  return arr.filter(FILTER_MAP[selectedOption]);
+}
```

Comment:
This idea with map is interesting, but you break filtering logic in several parts:

one is in `applyFilter` and another is inlined in the `App` component

You may keep the map and several methods, but I let's be consistent and put all parts of the filtering logic in one place

I recommend you placing them in a separate file, say `utils/todoFilter` or `utils/todoUtils`